### ISSUE TITLE
New G6L board (TUYA compatible version)

### ISSUE THREAD

**Issue by szymucha94** [pauln/esphome-linp-doorbell-g04#4]
Just bought the only available G6L doorbell in Poland.
![IMG_20221108_100300](https://user-images.githubusercontent.com/95378717/200522487-9d31aa65-72c8-40a6-b23e-7933bd16c8ae.jpg)
![IMG_20221108_100241](https://user-images.githubusercontent.com/95378717/200522503-5910bb23-ea32-4cfd-ac78-c40541a6857b.jpg)
![IMG_20221108_100250](https://user-images.githubusercontent.com/95378717/200522511-3a5d6f98-889a-417c-a532-85a5316206fb.jpg)

It seems that it has CBU TUYA-compatible MCU. Does ESPHome support CBU chips?

Cheers,
Szymon

**Comment by pauln** [comment]
From the [Tuya CBU datasheet](https://developer.tuya.com/en/docs/iot/cbu-module-datasheet?id=Ka07pykl5dk4u), it looks like it's a Beken BK7231N, which isn't (officially) supported by ESPHome at present.

There's a [draft PR to add support](https://github.com/esphome/esphome/pull/3509) via the [LibreTuya](https://github.com/kuba2k2/libretuya) project; I'm not sure what the current status is, but if you read through the comments on that PR and the LibreTuya documentation, you might enough information to try flashing it (if you're willing to risk making the doorbell useless if it doesn't work... I'd suggest checking if it's possible to dump and restore the original firmware before you try flashing custom firmware, unless you don't mind completely breaking the doorbell if something goes wrong / it doesn't work yet).  You'd also need to determine which pins to use for the `remote_receiver` and `remote_transmitter` configuration, as I don't expect they'd match the ESP32 pins.

**Comment by szymucha94** [comment]
You know what... I just bought zigbee illuminance sensor to report doorbell's LED activity to HA. I guess it's the easiest way to get what I need :)
I can just mention that I've captured traffic generated by the base and it's all encrypted. Tuya API browser also doesn't report doorbell's local key. I guess at this moment there are no dummy cloud projects that would allow keeping the wifi connection up and decryption of this traffic to work locally with the stock firmware.

**Comment by pcmester** [comment]
> You know what... I just bought zigbee illuminance sensor to report doorbell's LED activity to HA. I guess it's the easiest way to get what I need :) I can just mention that I've captured traffic generated by the base and it's all encrypted. Tuya API browser also doesn't report doorbell's local key. I guess at this moment there are no dummy cloud projects that would allow keeping the wifi connection up and decryption of this traffic to work locally with the stock firmware.

You can get full local control of the device with the localtuya HA integration, this works way better than with the Xiaomi Cloud-tied board, here are the props you can get/set via the cloud API and locally:

 ```
"functions": [
      {
        "code": "doorbell_ring_value",
        "dp_id": 2,
        "type": "Integer",
        "values": "{\"unit\":\"\",\"min\":1,\"max\":40,\"scale\":0,\"step\":1}"
      },
      {
        "code": "doorbell_volume_value",
        "dp_id": 3,
        "type": "Integer",
        "values": "{\"unit\":\"\",\"min\":0,\"max\":8,\"scale\":0,\"step\":1}"
      },
      {
        "code": "alarm_propel_switch",
        "dp_id": 6,
        "type": "Boolean",
        "values": "{}"
      }
    ],
    "status": [
      {
        "code": "doorbell_ring_value",
        "dp_id": 2,
        "type": "Integer",
        "values": "{\"unit\":\"\",\"min\":1,\"max\":40,\"scale\":0,\"step\":1}"
      },
      {
        "code": "doorbell_volume_value",
        "dp_id": 3,
        "type": "Integer",
        "values": "{\"unit\":\"\",\"min\":0,\"max\":8,\"scale\":0,\"step\":1}"
      },
      {
        "code": "alarm_propel_switch",
        "dp_id": 6,
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "doorbell_call",
        "dp_id": 10,
        "type": "Integer",
        "values": "{\"min\":1,\"max\":255,\"scale\":0,\"step\":1}"
      }
    ]
```

To get the local key, you can use the Tuya API browser, by sending a Get device details in the API browser.

**Comment by szymucha94** [comment]
@pcmester I've tried that already and tuya api browser did not return local key. Unless something changed in recent firmware releases. Do you own such doorbell? Which FW do you have?

**Comment by szymucha94** [comment]
Anyway, just tested it myself. They must've changed something because local key is clearly available now. Just added the device to HA. I hope it doesn't disconnect from wifi every 15 minutes when blocked from WAN :)
(edit: nope, it doesn't disconnect. It works fully offline).
One more note about this Beken SoC: it has the infamous mechanism that results in wifi MAC address being changed every device reboot. Which makes wifi bl*cklisting hard as MAC is unpredictable. I hope that this project will eventually be built using OpenBeken (https://github.com/openshwprojects/OpenBK7231T_App)

Thanks @pcmester!

**Comment by pcmester** [comment]
> Anyway, just tested it myself. They must've changed something because local key is clearly available now. Just added the device to HA. I hope it doesn't disconnect from wifi every 15 minutes when blocked from WAN :) (edit: nope, it doesn't disconnect. It works fully offline). One more note about this Beken SoC: it has the infamous mechanism that results in wifi MAC address being changed every device reboot. Which makes wifi bl*cklisting hard as MAC is unpredictable. I hope that this project will eventually be built using OpenBeken (https://github.com/openshwprojects/OpenBK7231T_App)
> 
> Thanks @pcmester!

I am glad that I was able to help. Yes, the MAC thing was a terrible headache for me, I could not connect it via the app because of that, but apparently it does not change much, I have like 3 slightly different MAC for the device, but not more. Once it is connected to the AP, it keeps the same address (at least I hope so).

I haven't tried to set different chimes for different buttons, maybe the API cannot do that locally, and also I have no clue what the  `alarm_propel_switch` does, it seems unavailable to me from HA. At least volume control works, and of course, the button press gets reported as an integer, which the older, Xiaomi versions could not do without ESPHome. One thing that doesn't work for me is the chime settings, because if I set it in HA, it clearly changes it, but when I press the buttons, they don't have the chime what I applied from HA.

**Comment by szymucha94** [comment]
> > Anyway, just tested it myself. They must've changed something because local key is clearly available now. Just added the device to HA. I hope it doesn't disconnect from wifi every 15 minutes when blocked from WAN :) (edit: nope, it doesn't disconnect. It works fully offline). One more note about this Beken SoC: it has the infamous mechanism that results in wifi MAC address being changed every device reboot. Which makes wifi bl*cklisting hard as MAC is unpredictable. I hope that this project will eventually be built using OpenBeken (https://github.com/openshwprojects/OpenBK7231T_App)
> > Thanks @pcmester!
> 
> I am glad that I was able to help. Yes, the MAC thing was a terrible headache for me, I could not connect it via the app because of that, but apparently it does not change much, I have like 3 slightly different MAC for the device, but not more. Once it is connected to the AP, it keeps the same address (at least I hope so).
> 
> I haven't tried to set different chimes for different buttons, maybe the API cannot do that locally, and also I have no clue what the `alarm_propel_switch` does, it seems unavailable to me from HA. At least volume control works, and of course, the button press gets reported as an integer, which the older, Xiaomi versions could not do without ESPHome. One thing that doesn't work for me is the chime settings, because if I set it in HA, it clearly changes it, but when I press the buttons, they don't have the chime what I applied from HA.

alarm_propel_switch (when configured as binary_sensor) returns either true or false, where true means that someone clicked the battery-less button, and false means that button is unpressed. You can use it for automations.
![1](https://user-images.githubusercontent.com/95378717/216375358-b4b7fbaf-6c60-4c84-9550-7a6b7f0b6d7e.PNG)
![2](https://user-images.githubusercontent.com/95378717/216375366-803bd4d8-1fa5-4933-9d5f-f773f96e204b.PNG)

**Comment by pcmester** [comment]
> > > Anyway, just tested it myself. They must've changed something because local key is clearly available now. Just added the device to HA. I hope it doesn't disconnect from wifi every 15 minutes when blocked from WAN :) (edit: nope, it doesn't disconnect. It works fully offline). One more note about this Beken SoC: it has the infamous mechanism that results in wifi MAC address being changed every device reboot. Which makes wifi bl*cklisting hard as MAC is unpredictable. I hope that this project will eventually be built using OpenBeken (https://github.com/openshwprojects/OpenBK7231T_App)
> > > Thanks @pcmester!
> > 
> > 
> > I am glad that I was able to help. Yes, the MAC thing was a terrible headache for me, I could not connect it via the app because of that, but apparently it does not change much, I have like 3 slightly different MAC for the device, but not more. Once it is connected to the AP, it keeps the same address (at least I hope so).
> > I haven't tried to set different chimes for different buttons, maybe the API cannot do that locally, and also I have no clue what the `alarm_propel_switch` does, it seems unavailable to me from HA. At least volume control works, and of course, the button press gets reported as an integer, which the older, Xiaomi versions could not do without ESPHome. One thing that doesn't work for me is the chime settings, because if I set it in HA, it clearly changes it, but when I press the buttons, they don't have the chime what I applied from HA.
> 
> alarm_propel_switch (when configured as binary_sensor) returns either true or false, where true means that someone clicked the battery-less button, and false means that button is unpressed. You can use it for automations. ![1](https://user-images.githubusercontent.com/95378717/216375358-b4b7fbaf-6c60-4c84-9550-7a6b7f0b6d7e.PNG) ![2](https://user-images.githubusercontent.com/95378717/216375366-803bd4d8-1fa5-4933-9d5f-f773f96e204b.PNG)

Ohh, thank you very much! I have just realized that I don't have a binary sensor at all, and the integer based `doorbell_call ` cannot be used alone for automation.

**Comment by pcmester** [comment]
Interesting, it does not work for me, the `binary_sensor` is always unavailable. What firmware version have you set it up with? 3.3 or 3.4? And have you updated the device in the Tuya Smart app to the latest version?

**Comment by szymucha94** [comment]
wow, turns out I'm stupid :D
Just checked, and this switch controls/reflects app setting "APP message push". I got confused during the night since I just tested the button once and combined it with WAN access changes.
I will play with DID 10 tho.

**Comment by pcmester** [comment]
Yes, that switch changes the push notification setting. DID 10 sort of works, it shows the last button press (if you have multiple buttons, it changes accordingly), but if you have only one, then it is constant, so cannot be used for automation this way. I have tried with When the doorbell_call (DID10) status changes, but nothing happens, because of its constant nature.

Also, even though you can set the volume, and it seems like it changes it, but actually, it does not work, nor the alarm sound. It play the new volume and the new alarm, but when you trigger it with the button, it reflect the setting you had in the app.

**Comment by szymucha94** [comment]
what kinda worked is:
![3](https://user-images.githubusercontent.com/95378717/216391749-42054bfd-3d33-40dc-aa88-544b75be7ddf.PNG)

Simple automation can set this "last id call" to 255 every time doorbell base sets it to expected "1". This way value "1" can act as a trigger for other automations. I hope the device doesn't reset it from 255 without button press... maybe it won't if I disable WAN for it :)

**Comment by szymucha94** [comment]
after 15 minutes it still works.
![4](https://user-images.githubusercontent.com/95378717/216398280-7b7a66b3-82a4-465f-833e-ecb52a46b3a6.PNG)

**Comment by pcmester** [comment]
Thank you, this trick works really well. An another thing that can be done, is that we allow 0 as the value of the Last button ID, and therefore 255 is not needed, but we can set it to 0 which does not exists. Now, the question remains that how can the ring chime and volume setting be "enforced" by HA, and the button-tied chime and volume settings which options seemingly aren't offered by the API browser.

Edit: 0 won't work, the doorbell does not like it. After HA sets the state, the bell return the status as 1, so it changes back to 1 in HA too. 255 is fine.

**Comment by szymucha94** [comment]
it's risky, because MCU program may not expect it to be 0 (variable types?), so personally I'll leave it as 255

Edit: you were faster :)

**Comment by szymucha94** [comment]
Turns out this doorbell was not blocked from WAN... Standard rules that are rejecting all packets coming from specific MACs and IPs somehow did not work for this one device (maybe it uses multiple MACs in ethernet frames despite keeping one for active WiFi connection?). Now I've sniffed that doorbell talks to ec2-35-156-44-172.eu-central-1.compute.amazonaws.com and I've set this hostname to static non-existant IP in local network. I'm right now testing if doorbell will restart every 15 minutes like almost all IoT devices do these days. Hopefully not because it takes more than 40 seconds for it to reconnect which creates kinda big windows for it to not report in with button call. Not to mention that when HA initializes DPs it makes sound on the doorbell base ;)
Also increased TCP SYN and rejects are now generating useless traffic on wifi... it'd be still ideal to have ESPhome or some other replacement for it.

**Comment by pcmester** [comment]
> Turns out this doorbell was not blocked from WAN... Standard rules that are rejecting all packets coming from specific MACs and IPs somehow did not work for this one device (maybe it uses multiple MACs in ethernet frames despite keeping one for active WiFi connection?). Now I've sniffed that doorbell talks to ec2-35-156-44-172.eu-central-1.compute.amazonaws.com and I've set this hostname to static non-existant IP in local network. I'm right now testing if doorbell will restart every 15 minutes like almost all IoT devices do these days. Hopefully not because it takes more than 40 seconds for it to reconnect which creates kinda big windows for it to not report in with button call. Not to mention that when HA initializes DPs it makes sound on the doorbell base ;) Also increased TCP SYN and rejects are now generating useless traffic on wifi... it'd be still ideal to have ESPhome or some other replacement for it.

On my side, everything works without WAN access, no reboots so far.

**Comment by plantroon** [comment]
I definitely tried localtuya back in the day, and it definitely didn't work for me, I don't remember why. So it works now? I can have the doorbell locally on stock firmware without internet access? :)

I have 2 doorbells, one is still on stock firmware and I was using it to catch missed button presses of esphome-flashed receiver vs the stock firmware one. Everything works fine now, so this would be interesting to add one more doorbell to my HASS at some other place without destroying its case :D

**Comment by szymucha94** [comment]
> I definitely tried localtuya back in the day, and it definitely didn't work for me, I don't remember why. So it works now? I can have the doorbell locally on stock firmware without internet access? :)
> 
> I have 2 doorbells, one is still on stock firmware and I was using it to catch missed button presses of esphome-flashed receiver vs the stock firmware one. Everything works fine now, so this would be interesting to add one more doorbell to my HASS at some other place without destroying its case :D

yeah, it works just fine for 2 weeks now. Just get the local tuya key and configure it according to structure quoted above :)

**Comment by spupetic** [comment]
I am trying to make this custom component work on the CBU (BK7231N) running ESPHome via LibreTiny. Everything works, except the remote receiver. The log says, that `Remote Receiver Signal starts with a HIGH value. Usually this means you have to invert the signal using 'inverted: True' in the pin schema!`, but when I invert it, nothing changes, still no button presses in the log. I have tried setting the log level to very verbose, that does not help either.

**Comment by szymucha94** [comment]
> I am trying to make this custom component work on the CBU (BK7231N) running ESPHome via LibreTiny. Everything works, except the remote receiver. The log says, that `Remote Receiver Signal starts with a HIGH value. Usually this means you have to invert the signal using 'inverted: True' in the pin schema!`, but when I invert it, nothing changes, still no button presses in the log. I have tried setting the log level to very verbose, that does not help either.

Don't you have to pair remote button first after reflash?

**Comment by spupetic** [comment]
> > I am trying to make this custom component work on the CBU (BK7231N) running ESPHome via LibreTiny. Everything works, except the remote receiver. The log says, that `Remote Receiver Signal starts with a HIGH value. Usually this means you have to invert the signal using 'inverted: True' in the pin schema!`, but when I invert it, nothing changes, still no button presses in the log. I have tried setting the log level to very verbose, that does not help either.
> 
> Don't you have to pair remote button first after reflash?

The pairing process works using the dumper, I should be able to catch the button's ID, and then add it to the config. That part does not work for me though.

**Comment by szymucha94** [comment]
@spupetic did you happen to fix this issue? I'm considering flashing esphome to have the invalid MAC address fixed.

**Comment by spupetic** [comment]
> @spupetic did you happen to fix this issue? I'm considering flashing esphome to have the invalid MAC address fixed.

Unfortunately, no. And the problem is, that I don't really know, what the exact problem is. The doorbell chips work well under ESPHome from the Tuya board, if I connect them to an ESP-based board, not just the 32 but the 8266 too. So I guess that libretiny does something different than the original ESPHome, but I haven't seen any progress with this on that side.

**Comment by szymucha94** [comment]
> > @spupetic did you happen to fix this issue? I'm considering flashing esphome to have the invalid MAC address fixed.
> 
> Unfortunately, no. And the problem is, that I don't really know, what the exact problem is. The doorbell chips work well under ESPHome from the Tuya board, if I connect them to an ESP-based board, not just the 32 but the 8266 too. So I guess that libretiny does something different than the original ESPHome, but I haven't seen any progress with this on that side.

Meh, that's a bummer.
Which ESP32 did you use? Do you know pinout and which pins of cbu need to be connected to effectively replace mcu? I've got a spare esp-wroom-32d on hand, so I could use it.

**Comment by spupetic** [comment]
> > > @spupetic did you happen to fix this issue? I'm considering flashing esphome to have the invalid MAC address fixed.
> > 
> > 
> > Unfortunately, no. And the problem is, that I don't really know, what the exact problem is. The doorbell chips work well under ESPHome from the Tuya board, if I connect them to an ESP-based board, not just the 32 but the 8266 too. So I guess that libretiny does something different than the original ESPHome, but I haven't seen any progress with this on that side.
> 
> Meh, that's a bummer. Which ESP32 did you use? Do you know pinout and which pins of cbu need to be connected to effectively replace mcu? I've got a spare esp-wroom-32d on hand, so I could use it.

The remote receiver is on P8, the transmitter is P26 and the two LEDs are P28 and P7 on the CBU. Take a look at the LibreTiny documentation of the CBU board to make the connections. I simply soldered wires onto the CBUs exposed pads, I have not removed it. I have not used or tested the buttons. I think you can use the pins written in this code, but other can work as well.

**Comment by szymucha94** [comment]
> > > > @spupetic did you happen to fix this issue? I'm considering flashing esphome to have the invalid MAC address fixed.
> > > 
> > > 
> > > Unfortunately, no. And the problem is, that I don't really know, what the exact problem is. The doorbell chips work well under ESPHome from the Tuya board, if I connect them to an ESP-based board, not just the 32 but the 8266 too. So I guess that libretiny does something different than the original ESPHome, but I haven't seen any progress with this on that side.
> > 
> > 
> > Meh, that's a bummer. Which ESP32 did you use? Do you know pinout and which pins of cbu need to be connected to effectively replace mcu? I've got a spare esp-wroom-32d on hand, so I could use it.
> 
> The remote receiver is on P8, the transmitter is P26 and the two LEDs are P28 and P7 on the CBU. Take a look at the LibreTiny documentation of the CBU board to make the connections. I simply soldered wires onto the CBUs exposed pads, I have not removed it. I have not used or tested the buttons. I think you can use the pins written in this code, but other can work as well.

Thanks. I guess stealing 3.3V from doorbell's mobo for the new wroom while powering whole thing up with 230V was not such a brilliant idea. Killed doorbell, both MCUs and UART bridge :D
Ordered esp32 powered g6l. Hopefully it works stable with repo's program.
![IMG_20240723_005921](https://github.com/user-attachments/assets/17d5b31f-3ca7-4756-9d34-9b0d28908f4b)

**Comment by spupetic** [comment]
> > > > > @spupetic did you happen to fix this issue? I'm considering flashing esphome to have the invalid MAC address fixed.
> > > > 
> > > > 
> > > > Unfortunately, no. And the problem is, that I don't really know, what the exact problem is. The doorbell chips work well under ESPHome from the Tuya board, if I connect them to an ESP-based board, not just the 32 but the 8266 too. So I guess that libretiny does something different than the original ESPHome, but I haven't seen any progress with this on that side.
> > > 
> > > 
> > > Meh, that's a bummer. Which ESP32 did you use? Do you know pinout and which pins of cbu need to be connected to effectively replace mcu? I've got a spare esp-wroom-32d on hand, so I could use it.
> > 
> > 
> > The remote receiver is on P8, the transmitter is P26 and the two LEDs are P28 and P7 on the CBU. Take a look at the LibreTiny documentation of the CBU board to make the connections. I simply soldered wires onto the CBUs exposed pads, I have not removed it. I have not used or tested the buttons. I think you can use the pins written in this code, but other can work as well.
> 
> Thanks. I guess stealing 3.3V from doorbell's mobo for the new wroom while powering whole thing up with 230V was not such a brilliant idea. Killed doorbell, both MCUs and UART bridge :D Ordered esp32 powered g6l. Hopefully it works stable with repo's program. ![IMG_20240723_005921](https://private-user-images.githubusercontent.com/95378717/351138801-17d5b31f-3ca7-4756-9d34-9b0d28908f4b.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE3MTk5NzQsIm5iZiI6MTcyMTcxOTY3NCwicGF0aCI6Ii85NTM3ODcxNy8zNTExMzg4MDEtMTdkNWIzMWYtM2NhNy00NzU2LTlkMzQtOWIwZDI4OTA4ZjRiLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIzVDA3Mjc1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjMDU0NWE1YjViOTM1ZjEwODk1OWU0YjA3NGFjOTExZjBlYjlmYzM0ZTM3ZGJmNzBiZTYxMGZlZjNhNjQ1NTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2gtCT5UqTBgkSiYChJf8l9OVB5p73vze37mv_Armr_k)

I have been doing the tests from DC power, so I have not used the wall plug that times. I don't know what went wrong there, but getting a compatible board is certainly an easier, but more expensive route.

**Comment by strasharo** [comment]
Has anyone succeeded with this version?

![Image](https://github.com/user-attachments/assets/fb01f09c-360e-4b97-b7ed-41b36e7d2090)

![Image](https://github.com/user-attachments/assets/70a67e51-3275-486d-aca4-0616d2eef4f1)

**Comment by szymucha94** [comment]
Get a drop-in replacement for CBU like this one:
"ESP32-C3 Tuya CBU Replacement Module (ESP8685-WROOM-06)".

**Comment by strasharo** [comment]
@szymucha94 will this one do it?

https://vi.aliexpress.com/item/1005004537860775.html

**Comment by szymucha94** [comment]
> [@szymucha94](https://github.com/szymucha94) will this one do it?
> 
> https://vi.aliexpress.com/item/1005004537860775.html

yup
Do keep in mind you need to desolder CBU chip with hot air station, or at least gas powered heating gun. It won't work with just soldering iron.

**Comment by strasharo** [comment]
Yep, a friend of mine just ordered a this one, hopefully it'll do it:

![Image](https://github.com/user-attachments/assets/a4d7b1f0-ecef-4766-aee7-c46cc7672af5)

Thanks a lot for the tips. Just one last question - what about the pinout for reflashing afterwards, is it the same as the one listed in this repo?

**Comment by szymucha94** [comment]
It's definitely different. But traces are easily visible on the PCB so you can figure out which GPIO is connected to other components within minutes.
Also, some clues are here:
https://github.com/pauln/esphome-linp-doorbell-g04/issues/7

**Comment by strasharo** [comment]
Unfortunately I'm not that proficient in this part. :( I'll ask in the other thread.

**Comment by strasharo** [comment]
Hey @szymucha94 , sorry to bug you, but I just received the ESP8685-WROOM-06, could you, please, give a hint what would be the pinout for soldering it to the board? :) Old one is already off:

![Image](https://github.com/user-attachments/assets/6027c87e-2fab-4ab9-9d56-85a17bbd3829)

![Image](https://github.com/user-attachments/assets/31e0f76d-51cd-49e9-80b0-631f961080f8)

**Comment by szymucha94** [comment]
@strasharo This is merely my guess based on these pictures:

remote_receiver (check) - GPIO8
remote_transmitter (check) - GPIO11
blue_led (check) - GPIO9 or GPIO17
orange_led (check) - GPIO17 or GPIO9
physical button A (check) - GPIO10 or GPIO19
physical button B (check) - GPIO19 or GPIO10

translation above done via: [WROOM](https://templates.blakadder.com/ESP8685-WROOM-06.html) and [CBU](https://docs.libretiny.eu/boards/cbu/#pinout)

With CBU pin scheme:
P14(unused, looks like support joint for MCU)
P16(unused, looks like support joint for MCU)
P1(unused? seems like a test pad)
P0(unused? seems like a test pad)
P8(remote_receiver)
P7(LED, blue/orange?)
P6(physical button, left)
P26(remote_transmitter?)
P24(? SZY8039B out for LED? unused then)
GND(doesn't matter)
3V3(doesn't matter)
P11(doesn't matter)
P10(doesn't matter)
P28(LED, blue/orange?)
CEN(doesn't matter)
P9(physical button, right)
P15(unused, looks like support joint for MCU)

P21(unused, on the back of MCU)

Proceed with caution. Best to confirm this with multimeter. Or if you like the risk try first just the remote_receiver and transmitter, leave LEDs and buttons unset. I can barely see traces on the white paint area.

**Comment by strasharo** [comment]
Thanks a lot @szymucha94 . Here are the closeups of the white paint area:

![Image](https://github.com/user-attachments/assets/f1648aa4-ca2c-4033-af7d-ca6c894637ff)

![Image](https://github.com/user-attachments/assets/41c9cfad-de20-421d-809a-8cad5b5e8b9f)

![Image](https://github.com/user-attachments/assets/ef5a7391-1c6f-47c1-bd7e-4f06be92a6a9)

**Comment by julianczaja** [comment]
I've successfully replaced the original CBU module with an **ESP8685-WROOM-06** in the Linptech G04. Here is the pinout for ESPHome:
- remote_receiver: GPIO3
- remote_transmitter: GPIO4
- orange_led: GPIO7
- blue_led: GPIO19*
- left button: GPIO6
- right button: GPIO10

\* The original blue LED is connected to pin 17 of the module footprint, but on the ESP8685-WROOM-06 pin 17 is marked as NC (Not Connected). To make it work, I cut the PCB trace from the LED to pin 17 and soldered a small wire directly from the LED to pin 16 (GPIO19).

**Comment by szymucha94** [comment]
Btw, there is no need to replace CBUs anymore. They're officially supported by esphome and they work nicely.

**Comment by julianczaja** [comment]
Oh, good to know, thanks ;)

**Comment by spupetic** [comment]
> Btw, there is no need to replace CBUs anymore. They're officially supported by esphome and they work nicely.

How did you make the CBU version work with ESPHome?

**Comment by szymucha94** [comment]
> > Btw, there is no need to replace CBUs anymore. They're officially supported by esphome and they work nicely.
> 
> How did you make the CBU version work with ESPHome?

Normally. It is supported by esphome out-of-the-box. From one of my sensors:

```
esphome:
  name: pir-cbu
  friendly_name: PIR CBU

bk72xx:
  board: cbu
```

If you're asking in context of this project - just figure out pins on the board and modify repo's yaml. I got couple of CBU chips around my place all running esphome. No issues so far.

**Comment by spupetic** [comment]
> > > Btw, there is no need to replace CBUs anymore. They're officially supported by esphome and they work nicely.
> > 
> > 
> > How did you make the CBU version work with ESPHome?
> 
> Normally. It is supported by esphome out-of-the-box. From one of my sensors:
> 
> ```
> esphome:
>   name: pir-cbu
>   friendly_name: PIR CBU
> 
> bk72xx:
>   board: cbu
> ```
> 
> If you're asking in context of this project - just figure out pins on the board and modify repo's yaml. I got couple of CBU chips around my place all running esphome. No issues so far.

Yeah, the CBU itself is supported, and the chime chip works, but the remote receiver does not, which means that button presses cannot be received with ESPHome firmware.

**Comment by szymucha94** [comment]
> > > > Btw, there is no need to replace CBUs anymore. They're officially supported by esphome and they work nicely.
> > > 
> > > 
> > > How did you make the CBU version work with ESPHome?
> > 
> > 
> > Normally. It is supported by esphome out-of-the-box. From one of my sensors:
> > ```
> > esphome:
> >   name: pir-cbu
> >   friendly_name: PIR CBU
> > 
> > bk72xx:
> >   board: cbu
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > If you're asking in context of this project - just figure out pins on the board and modify repo's yaml. I got couple of CBU chips around my place all running esphome. No issues so far.
> 
> Yeah, the CBU itself is supported, and the chime chip works, but the remote receiver does not, which means that button presses cannot be received with ESPHome firmware.

Have you seen this thread?
https://github.com/libretiny-eu/libretiny/issues/69
Which version of libretiny are you using? You can try forcing the newest framework version in esphome config.
